opslevel/terraform_type_conversions.go

```diff
@@ -24,6 +24,14 @@ func OptionalStringValue(value string) basetypes.StringValue {
 	return types.StringValue(unquote(value))
 }
 
+// Returns the config value as a NullableValue[string]. If the value is not set, null, or "" in the config, the value will be treated as null.
+func NullableStringConfigValue(s types.String) *opslevel.NullableValue[string] {
```

Comment:
This part is wrong,

>or "" in the config

Terraform and opslevel-go can't detect if a service description, framework, etc. are unset or just `""` (since these fields are concrete types in `Service`) and so you can't actually set fields to `""` in terraform...

```
  # opslevel_service.mon_service will be created
  + resource "opslevel_service" "mon_service" {
      + framework       = "rails"
      + id              = (known after apply)
      + language        = "ruby"
      + last_updated    = (known after apply)
      + lifecycle_alias = "pre-alpha"
      + name            = "mon_service"
      + owner           = "team_platform_3"
      + tier_alias      = "tier_4"
        # (1 unchanged attribute hidden)
    }

Plan: 1 to add, 0 to change, 0 to destroy.
opslevel_service.mon_service: Creating...
╷
│ Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to opslevel_service.mon_service, provider "provider[\"registry.terraform.io/opslevel/opslevel\"]" produced an unexpected new value: .description: was cty.StringVal(""), but now null.
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵

```

Should we care?

Reply:
Should be fixed in a recent commit, can unresolve this later if it's still a prob